ui/repl.c

```diff
@@ -186,9 +187,14 @@ void parse_opts(int *argcp, char ***argvp)
             break;
         case 'p': // procs
             errno = 0;
-            jl_options.nprocs = strtol(optarg, &endptr, 10);
-            if (errno != 0 || optarg == endptr || *endptr != 0 || jl_options.nprocs < 1)
-                jl_errorf("julia: -p,--procs=<n> must be an integer >= 1\n");
+            if (!strcmp(optarg,"auto")) {
+                jl_options.nprocs = jl_cpu_cores();
```

Comment:
It refers to additional worker processes. Since the master will be the driver and not used for computation, it makes sense to fire up as many workers as cores.